tests/rules/test_quality.py

```
@@ -79,3 +80,34 @@ def test_frame_summary_and_rule_skips_all_sentinel_required():
     assert fq.roles["fan-status"].valid_coverage == 1
     result = run_rule("VAV-1", df, poll_seconds=300.0, require_operational_gates=False)
     assert result.status == "SKIPPED_MISSING_ROLES"
+
+
+def test_apply_normalized_does_not_fragment():
+    idx = _idx()
+    df = pd.DataFrame(
+        {f"zone-air-temp": [72.0] * 6, "fan-status": [1.0] * 6},
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)